src/org/exist/xquery/functions/util/LogFunction.java

```diff
@@ -3,191 +3,213 @@
-                        buf.append(getSource().getKey());
-                    }
+public class LogFunction extends BasicFunction {
+
+    protected static final FunctionParameterSequenceType PRIORITY_PARAMETER
+            = new FunctionParameterSequenceType("priority", Type.STRING, Cardinality.EXACTLY_ONE, "The logging priority: 'error', 'warn', 'debug', 'info', 'trace'");
+    protected static final FunctionParameterSequenceType LOGGER_NAME_PARAMETER
+            = new FunctionParameterSequenceType("logger-name", Type.STRING, Cardinality.EXACTLY_ONE, "The name of the logger, eg: my.app.log");
+    protected static final FunctionParameterSequenceType MESSAGE_PARAMETER
+            = new FunctionParameterSequenceType("message", Type.ITEM, Cardinality.ZERO_OR_MORE, "The message to log");
+
+    protected static final Logger logger = LogManager.getLogger(LogFunction.class);
+
+    public static final String FUNCTION_LOG = "log";
+    public static final String FUNCTION_LOGAPP = "log-app";
+    public static final String FUNCTION_LOG_SYSTEM_OUT = "log-system-out";
+    public static final String FUNCTION_LOG_SYSTEM_ERR = "log-system-err";
+    public final static FunctionSignature signatures[] = {
+            new FunctionSignature(
+                    new QName(FUNCTION_LOG, UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Logs the message to the current logger.",
+                    new SequenceType[]{PRIORITY_PARAMETER, MESSAGE_PARAMETER},
+                    new SequenceType(Type.ITEM, Cardinality.EMPTY)
+            ),
+            new FunctionSignature(
+                    new QName(FUNCTION_LOG_SYSTEM_OUT, UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Logs the message to System.out.",
+                    new SequenceType[]{MESSAGE_PARAMETER},
+                    new SequenceType(Type.ITEM, Cardinality.EMPTY)
+            ),
+            new FunctionSignature(
+                    new QName(FUNCTION_LOG_SYSTEM_ERR, UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Logs the message to System.err.",
+                    new SequenceType[]{MESSAGE_PARAMETER},
+                    new SequenceType(Type.ITEM, Cardinality.EMPTY)
+            ),
+            new FunctionSignature(
+                    new QName(FUNCTION_LOGAPP, UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Logs the message to the named logger",
+                    new SequenceType[]{PRIORITY_PARAMETER, LOGGER_NAME_PARAMETER, MESSAGE_PARAMETER},
+                    new SequenceType(Type.ITEM, Cardinality.EMPTY)
+            )
+    };
+
+    public LogFunction(XQueryContext context, FunctionSignature signature) {
+        super(context, signature);
+    }
+
+    /* (non-Javadoc)
+     * @see org.exist.xquery.BasicFunction#eval(org.exist.xquery.value.Sequence[], org.exist.xquery.value.Sequence)
+     */
+    public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathException {
```

Comment:
The junk Eclipse javadoc comment should be replaced with `@Override` whilst you are cleaning up.